feat: v3 fee bot (dry run only)

gosuto-inzasheru opened the pull request:
dry run, so the bot doesnt actually broadcast the calls to the sweeper yet

sample output:
```
checking for pending fees on sepolia:0x9c5c0bb79af59c0fb4624330eb3642eb2dbf8911 (DAI-USDC)...
checking for pending fees on sepolia:0x2766595b4fa4440d2edf5f987556673743b085fa (B-80VAULTX-20WETH)...
checking for pending fees on sepolia:0xc7512572bca89b90d604f88557d270716dcfea78 (Bv3-DAI-USDC-DirectionalFee)...
checking for pending fees on sepolia:0xaf4dd6e7a6dddb6df29b5425aeb4cf2c499aedbc (50USDC-50BALLISH)...
token: BAL 0xb19382073c7a0addbb56ac6af1808fa49e377b75
price: $1.4499999999999999555910790149937383830547332763671875
collectable in vault: 0.0015 BAL
sweepable in controller: 0 BAL
burning for: 0.002174999999999999933386618522 USDC
tx: {'chainId': 11155111,
 'data': '0xb9746995000000000000000000000000af4dd6e7a6dddb6df29b5425aeb4cf2c499aedbc000000000000000000000000b19382073c7a0addbb56ac6af1808fa49e377b7500000000000000000000000000000000000000000000000000000000000003e80000000000000000000000000000000000000000000000000000000067dd629a0000000000000000000000003400aafa47c6dc0d8ffc2380d3665f701cd53ebc',
 'from': '0x74E283B985EA76c55C8B48d6bD1067a418188424',
 'gas': 404015,
 'maxFeePerGas': 206856677,
 'maxPriorityFeePerGas': 1003477,
 'nonce': 4,
 'to': '0x5E69888b605a181A5615aC1b3C8937aFA2913d15',
 'value': 0}
tx hash: <dry run>

checking for pending fees on sepolia:0x4c4e962b797dc631f6e022308f07b1b4438beb4d (69BAL-31BEETS)...
checking for pending fees on sepolia:0x395c83c812e4222aba083d3f3dc7c82e30d61124 (50WETH-50LINK)...
```

Xeonus commented on action-scripts/v3_fee_bot.py:
```diff
@@ -0,0 +1,136 @@
+
+
+V3_CHAINS = ["sepolia"] + AddrBook("mainnet").chains.BALANCER_PRODUCTION_CHAINS_V3
+USDC_THRESHOLD = 0.001
```
We might want to move this constant to a general config file once this goes to PROD?

Xeonus commented on action-scripts/v3_fee_bot.py:
```diff
@@ -0,0 +1,136 @@
+                        print("!!! burner not approved (yet); skipping\n")
+                        continue
+                    try:
+                        unsigned_tx = (
```
What happens if a tx fails? I guess it is OK there is no check as there will simply be another broadcast later?

Xeonus replied:
What happens if  we have a gas spike upon submitting a tx? Can block the entire queue?

gosuto-inzasheru replied:
it is in a `try` here because `build_transaction` sims it first (to estimate gas). if that reverts, we catch it

still some better handling of edge cases should be thought about further, i agree

Xeonus reviewed (commented):
- core logic checks out
- dry run successful for sepolia (without broadcasting anything yet)
- before approva, have general comments how to do error handling around broadcasting txs
- Otherwise very clean code

gosuto-inzasheru commented:
@Xeonus i moved your concerns to https://github.com/BalancerMaxis/multisig-ops/issues/1846 (and added some i already had)

part of the reason for merging this is to do a full dry run in prod and learn from that. as long as `broadcast` is kept false, all this is just testing in prod

Xeonus reviewed (approved):
Error handling and flow improvements will be done in separate PR. Approving this now so dry-runs can be tested. Core logic is solid
